src/target/riscv/riscv-013.c

```diff
@@ -1636,32 +1712,84 @@ static int read_memory(struct target *target, target_addr_t address,
 	dmi_write(target, DMI_ABSTRACTAUTO, 0);
 
 	if (count > 1) {
-		/* Read the penultimate word. */
+		// Read the penultimate word.
 		uint64_t value = dmi_read(target, DMI_DATA0);
 		write_to_buf(buffer + receive_addr - address, value, size);
 		LOG_DEBUG("M[0x%" TARGET_PRIxADDR "] reads 0x%" PRIx64, receive_addr, value);
 		receive_addr += size;
 	}
 
-	/* Read the last word. */
+	// Read the last word.
 	uint64_t value;
-	result = register_read_direct(target, &value, GDB_REGNO_S1);
-	if (result != ERROR_OK)
-		goto error;
+	if (register_read_direct(target, &value, GDB_REGNO_S1) != ERROR_OK)
+		return ERROR_FAIL;
 	write_to_buf(buffer + receive_addr - address, value, size);
 	LOG_DEBUG("M[0x%" TARGET_PRIxADDR "] reads 0x%" PRIx64, receive_addr, value);
 	receive_addr += size;
 
 	riscv_set_register(target, GDB_REGNO_S0, s0);
 	riscv_set_register(target, GDB_REGNO_S1, s1);
-	return ERROR_OK;
+	}
+	else{ //System Bus Access
+		LOG_DEBUG("*** read_memory System Buss Access: size: %d\tcount:%d\tstart address: 0x%08" PRIx64,size,count,address);
+		uint8_t *t_buffer=buffer;
+		riscv_addr_t cur_addr = address;
+		riscv_addr_t fin_addr = address + (count * size);
+		uint64_t access=0;
+
+		//ww favorise one off reading if there is an issu
+		if(count==1){  
+			for(uint32_t i=0; i<count ; i++){
+				access= dmi_read(target,DMI_SBCS);
+				dmi_write(target, DMI_SBADDRESS0, cur_addr);
+				//size/2 matching the bit access of the spec 0.13
+					,DMI_SBCS_SBACCESS,size/2); 
```

Comment:
What's the intent here? I was trying to clean up this PR, but this new code doesn't even compile.

Reply:
:(....
access=set_field(access,DMI_SBCS_SBACCESS,size/2);

Reply:
sorry...